srcs/http/request/parse/http_parse.cpp

```diff
@@ -32,48 +36,65 @@ bool IsStringUpper(const std::string &str) {
 	return true;
 }
 
+// OWS (optional whitespace): SP(Space), HTAB(Horizontal Tab)
+void TrimLeadingOptionalWhitespace(std::string &str) {
+	typedef std::string::iterator It;
+	for (It it = str.begin(); it != str.end(); ++it) {
+		if (!IsOptionalWhitespace(*it)) {
+			str.erase(str.begin(), it);
+			return;
+		}
+	}
+	str.erase(str.end());
```

Comment:
全部消すっていう意味で合っていれば、`clear()` が向いてるかもです
```suggestion
	str.clear();
```

Reply:
後で知ったのですが、こういう方法もありました
これだと `IsOptionalWhitespace()` が使えないので微妙かもですが、参考までに共有です 
```cpp
void TrimLeadingOptionalWhitespace(std::string &str) {
  static const std::string OPTIONAL_WHITESPACE = SP + HTAB;
  std::string::size_type   pos                 = str.find_first_not_of(OPTIONAL_WHITESPACE);
  if (pos != std::string::npos) {
  	str.erase(0, pos);
  } else {
  	str.clear();
  }
}
```

また、trim 関数が string を参照で受け取り、内部で erase() することへの違和感は、以下の cpp でよく使われるものが全部 string を返すからでした。言語化できずすみません…
```cpp
std::string s  = "abcde";
std::string s2 = s.erase(0, 2);　 // cde
std::string s3 = s2.substr(0, 2);  // cd
std::string s4 = std::string(s3.begin() + 1, s3.end()); // d
// ついでに libft の strtrim なども
```